aspnetcore/fundamentals/http-context.md

```diff
@@ -0,0 +1,146 @@
+
+ASP.NET Core applications access the HttpContext through the [IHttpContextAccessor](/dotnet/api/microsoft.aspnetcore.http.ihttpcontextaccessor?view=aspnetcore-2.0) interface and its default implementation [HttpContextAccessor](/dotnet/api/microsoft.aspnetcore.http.httpcontextaccessor.httpcontext?view=aspnetcore-2.0).
+
+## Use the HttpContext from Razor Pages, controllers, and middleware
+
+The Razor Pages `PageModel` exposes the `HttpContext` property:
+
+```csharp
+public class AboutModel : PageModel
+{
+    public string Message { get; set; }
+
+    public void OnGet()
+    {
+        Message = HttpContext.Request.PathBase;
+    }
+}
+```
+
+Controllers expose the `HttpContext` property from the `ControllerBase`:
+
+```csharp
+public IActionResult About()
+{
+    var pathBase = HttpContext.Request.PathBase;
+    // Do something with the PathBase.
+
+    return View();
+}
+```
+
+When working with custom middleware components, `HttpContext` is passed into the `Invoke` method and can be accessed when the middleware is being configured.
+
+```csharp
+public class MyCustomMiddleware
+{
+    public async Task Invoke(HttpContext context)
+    {
+        // Middleware initialization optionally using HttpContext
+    }
+    ...
+}
+```
+
+## Use HttpContext from custom components
+
+For other framework and custom components that need access to `HttpContext`, the recommended approach is to register a dependency using the built-in [dependency injection](xref:fundamentals/dependency-injection) container. The dependency injection container will supply the `IHttpContextAccessor` to any classes that declare it as a dependency in their constructors.
+
+```csharp
+public void ConfigureServices(IServiceCollection services)
+{
+     services.AddMvc();
+     services.AddSingleton<IHttpContextAccessor, HttpContextAccessor>();
+     services.AddTransient<IUserRepository, UserRepository>();
+}
+```
+
+In the preceding example:
+
+* `UserRepository` declares its dependency on `IHttpContextAccessor`.
+* The dependency is supplied when dependency injection resolves the dependency chain and creates an instance of `UserRepository`.
+
+```csharp
+public class UserRepository : IUserRepository
+{
+    private readonly IHttpContextAccessor _httpContextAccessor;
+
+    public UserRepository(IHttpContextAccessor httpContextAccessor)
+    {
+        _httpContextAccessor = httpContextAccessor;
+    }
+
+    public void LogCurrentUser()
+    {
+        var username = _httpContextAccessor.HttpContext.User.Identity.Name;
+        service.LogAccessRequest(username);
+    }
+
+    ...
+```
+
+## Work with libraries and ported code
+
+Registering IHttpContextAccessor in the application startup using the built-in dependency injection framework is the preferred approach to accessing HttpContext. In scenarios where it is not feasible to refactor to this approach, it is possible to simulate the behavior of `System.Web.HttpContext` from ASP.NET 4.x.
```

Comment:
I wouldn't say this at all. I would say it's required if you want to access the HttpContext inside of services that cannot otherwise get access to it.